osquery/filesystem/filesystem_tests.cpp

```diff
@@ -69,12 +69,38 @@ TEST_F(FilesystemTests, test_wildcard_full_recursion) {
             files.end());
 }
 
+TEST_F(FilesystemTests, test_wildcard_end_last_component){
+  std::vector<std::string> files;
+  auto status = resolveFilePattern("/%bin/%sh", files);
+  EXPECT_TRUE(status.ok());
+  EXPECT_NE(std::find(files.begin(), files.end(), "/bin/bash"),
+            files.end());
+  EXPECT_NE(std::find(files.begin(), files.end(), "/bin/sh"),
+            files.end());
+}
+
+TEST_F(FilesystemTests, test_wildcard_three_kinds){
+  std::vector<std::string> files;
+  auto status = resolveFilePattern("/%sr/%/%%", files);
```

Comment:
They can be replaced once we have gmock. This test checks that all wildcards can be chained together exactly as you'd expect them to be able to be. Is there a plan on gmock? or is there another way than both we should consider?